src/OrchardCore.Cms.Web/appsettings.json

```diff
@@ -112,6 +111,44 @@
     //      "RequestUrlPrefix": "tenant"
     //    }
     //  ]
-    //}
+    // },
+    "OrchardCore_Features": {
+      "Rules:100": {
+          "Rule": "Exclude",
```

Comment:
@jtkech when you have a minute might get some thoughts from you on this.

It all works fine (build just fails because of this testing code)

The problem I have is trying to get some ui in play for it, because `ShellSettings` only supports writable string keys and values.

Probably what I want to be able to do is write `IConfiguration` keys to the tenants `appsettings.json`. The format `"OrchardCore_Features:Rules:0:Rule": "Include",` would work, but when used as a value for shell settings, gets ignore because `OrchardCore_Features:Rules` creates a section with children, and here
https://github.com/OrchardCMS/OrchardCore/blob/953203f8b4a27caf3112bfb28012a28cf23d14b5/src/OrchardCore/OrchardCore/Shell/ShellSettingsManager.cs#L193

we ignore any keys with children.

So it might be a case of clearing all children in this section, and rewriting them with the values from the ui.

Here also in the standard appsettings.json file you will see I'm using a little of that format, to get a good array merge between different sections, due to how `IConfiguratonSection` handles arrays.

Better to reach out as you have more knowledge of this `UpdateableDataProvider` so may have some advice.

Reply:
Yes in `SaveSettingsAsync()` we make the distinction between tenant settings and configurations and we save both the settings in the source shared by all tenants (e.g. tenants.json) and the specific source (e.g. appsettings under the tenant folder). All data come from settings / config sources and here updated ones (settings or configs) are hold by the shell settings object through its `UpdateableDataProvider`, but when saving the settings we only take into account the well known fields required for starting a tenant, then other possible config updates are saved to the tenant specific config source.

Yes it is a sensitive area, normally a config source is immutable, we made 2 exceptions for the shared tenants settings and each tenant specific config sources for which we have an UI editor but only for a limited set o well known fields. In your case, because there are not tenant settings it would be saved as tenant configs. As i remember, i only implemented first level chidrens just because we don't have the UI to display all possible data, so if someone add data manually in the source they are not displayed, so to not be confusing i just ignored them both on editing and then on saving. Yes, now i remember, i already thought about it, having another UI to edit / update more nested data, hmm but yes when editing and saving a tenant it will clear these childrens. Hmm, yes maybe we just need to keep them into account, so that another UI tool could edit them, but as said it is sensite, this tool should not corrupt the required tenant config values, but why not.

Didn't have enough time to look at it more in details, i will think about it, ping me again if i forget ;)